src/execution_plan/optimizations/compact_filters.c

```diff
@@ -31,7 +31,7 @@ static void _removeTrueFilter(ExecutionPlan *plan, OpBase *op) {
 	ASSERT(root->t == FT_N_EXP);
 	// Evaluate the expression, and check if it is a 'true' value.
 	SIValue bool_val = AR_EXP_Evaluate(root->exp.exp, NULL);
-	if(SI_TYPE(bool_val) != T_BOOL) {
+	if(SI_TYPE(bool_val) != T_BOOL && SI_TYPE(bool_val) != T_NULL) {
```

Comment:
what should happen if `bool_val` is `NULL` ?

Reply:
you can't know here it depends on context